tests/test_pydeseq2.py

```diff
@@ -446,3 +446,29 @@ def test_vst(tol=0.02):
     assert (
         np.abs(r_vst_with_design - dds.layers["vst_counts"]) / r_vst_with_design
     ).max().max() < tol
+
+
+def test_ref_level():
+    """Test that DeseqDataSet columns are created according to the passed reference
+    level, if any.
+    """
+    counts_df = load_example_data(
+        modality="raw_counts",
+        dataset="synthetic",
+        debug=False,
+    )
+
+    clinical_df = load_example_data(
+        modality="clinical",
+        dataset="synthetic",
+        debug=False,
+    )
+
+    dds = DeseqDataSet(
+        counts=counts_df,
+        clinical=clinical_df,
+        design_factors=["group", "condition"],
+        ref_level=["group", "Y"],
+    )
+
+    assert "group_X_vs_Y" in dds.obsm["design_matrix"].columns
```

Comment:
do you think it might be worth for checking if it is correct? I still accept and leave the decision to you

Reply:
You mean checking whether the column contains correct data? It's true that this only checks that it exists.

Reply:
Yes, that's what I was thinking about. Would it be difficult to add?